conda-store-server/conda_store_server/action/download_packages.py

```diff
@@ -38,4 +75,96 @@ def action_fetch_and_extract_conda_packages(
                     ) = conda_package_streaming.url.conda_reader_for_url(url)
                     with file_path.open("wb") as f:
                         shutil.copyfileobj(conda_package_stream, f)
-                    conda_package_handling.api.extract(str(file_path))
+                    conda_package_handling.api.extract(file_path_str)
+
+                    # This code is needed to avoid failures when building in
+                    # parallel while using the shared cache.
+                    #
+                    # There are tarballs that only have the info/index.json file
+                    # and no info/repodata_record.json. The latter is used to
```

Comment:
I think `repodata_record.json` is _never_ in the remote tarball. It's actually created by `conda` upon extraction, taking `info.json` as the source, and extended with the actual info in `repodata.json`. See https://github.com/conda/conda/blob/6ec61a8445ae2663b134f8956104007827dafeab/conda/core/path_actions.py#L1430-L1441

Reply:
Tested locally with an inverted condition just in case. That didn't find any matching packages, so removed this check.